assets/javascripts/discourse/widgets/discourse-reactions-actions.js.es6

```diff
@@ -345,59 +347,97 @@ export default createWidget("discourse-reactions-actions", {
         scaleReactionAnimation(icon, scales[1], scales[0], () => {
           this.dropUserReaction();
-          this.addUserReaction(this.siteSettings.discourse_reactions_like_icon);
+          if (attrs.reaction && attrs.reaction !== mainReactionName) {
+            this.addUserReaction(attrs.reaction);
+          } else {
+            this.addUserReaction(
+              this.siteSettings.discourse_reactions_like_icon
+            );
+          }
 
-          if (!this.attrs.post.current_user_reaction) {
+          if (this.attrs.post.current_user_reaction) {
+            this.attrs.post.reaction_users_count -= 1;
+            this.setCurrentUserReaction();
+            this.attrs.post.current_user_used_main_reaction = false;
+          } else {
             this.attrs.post.reaction_users_count += 1;
             this.setCurrentUserReaction(
               this.siteSettings.discourse_reactions_like_icon
             );
             this.attrs.post.current_user_used_main_reaction = true;
-          } else {
-            this.attrs.post.reaction_users_count -= 1;
-            this.setCurrentUserReaction();
-            this.attrs.post.current_user_used_main_reaction = false;
           }
 
-          CustomReaction.toggle(
+          this.toggleCustomReaction(
             this.attrs.post.id,
-            this.siteSettings.discourse_reactions_reaction_for_like
-          ).then(value => {
-            resolve();
-            if (value === undefined) {
-              const mainReactionIcon = this.siteSettings
-                .discourse_reactions_like_icon;
-              const hasUsedMainReaction = this.attrs.post
-                .current_user_used_main_reaction;
-              const template = document.createElement("template");
-              template.innerHTML = iconHTML(
-                hasUsedMainReaction
-                  ? `far-${mainReactionIcon}`
-                  : mainReactionIcon
-              ).trim();
-              const mainReaction = template.content.firstChild;
-              icon.parentNode.replaceChild(mainReaction, icon);
-
-              this.attrs.post.current_user_reaction = current_user_reaction;
-              this.attrs.post.current_user_used_main_reaction = current_user_used_main_reaction;
-              this.attrs.post.reactions = reactions;
-              this.attrs.post.reaction_users_count = reaction_users_count;
-              this.scheduleRerender();
-            }
-          });
+            attrs.reaction && attrs.reaction !== mainReactionName
+              ? attrs.reaction
+              : this.siteSettings.discourse_reactions_reaction_for_like,
+            resolve,
+            {
+              current_user_reaction,
+              current_user_used_main_reaction,
+              reactions,
+              reaction_users_count
+            },
+            attrs,
+            true
+          );
         });
       });
     });
   },
 
+  toggleCustomReaction(
```

Comment:
really not liking this function, this is a monster, can you try to use composition to make it more understandable ?

Reply:
It's my fault 😅 I did this change for other reason, but that didn't work and I forgot to remove it